### ISSUE TITLE
Cache should not expire entites when Duration value is 0 (zero)

### ISSUE THREAD

**Issue by alparslanavci** [hazelcast/hazelcast#8148]
According to the JCache API javadoc, when the durationAmount parameter in ` javax.cache.expiry.Duration` constructor is set to 0 (zero), it means cache entites should live eternal. Here is the javadoc:

```
  /**
   * Constructs a duration.
   *
   * @param timeUnit       the unit of time to specify time in. The minimum time unit is milliseconds.
   * @param durationAmount how long, in the specified units, the cache entries should live. 0 means eternal.
   * @throws NullPointerException     if timeUnit is null
   * @throws IllegalArgumentException if durationAmount is less than 0 or a TimeUnit less than milliseconds is specified
   */
  public Duration(TimeUnit timeUnit, long durationAmount) {
```

However the entity directly expires when it is set to zero. See the test below. It fails when executed:

```
    @Test
    public void valueShouldNotBeExpiredWhenDurationIsZero() {
        final int CREATED_EXPIRY_TIME_IN_MSEC = 0;

        Duration duration = new Duration(TimeUnit.MILLISECONDS, CREATED_EXPIRY_TIME_IN_MSEC);
        CacheConfig<Integer, String> cacheConfig = new CacheConfig<Integer, String>();
        cacheConfig.setExpiryPolicyFactory(CreatedExpiryPolicy.factoryOf(duration));

        final Cache<Integer, String> cache = createCache(cacheConfig);
        cache.put(1, "value");

        assertTrueEventually(new AssertTask() {
            @Override
            public void run() throws Exception {
                assertNotNull(cache.get(1));
            }
        });
    }
```

**Comment by serkan-ozal** [comment]
I have looked at the implementation of `javax.cache.expiry.Duration` and noticed the expiration time can be interpreted as eternal if and only if time unit is `null` **and** duration time is `0`. So if duration time is `0` but time unit is not `null` (for example `MILLISECONDS`), it is **not** accepted as eternal. 

Regarding to the JCache TCK tests, `Duration ZERO = new Duration(SECONDS, 0)` means expiry immediately (not eternal because in here time unit is not `null` but `SECONDS`). 

So in fact there was no issue in our implementation. But yes, behaviour of duration time when it is `0` is not obviously clear from javadoc of jcache.

**Comment by jerrinot** [comment]
@serkan-ozal: this is a good realization! Should we raise a ticket in the JSR-107 project?

**Comment by serkan-ozal** [comment]
@jerrinot To me javadoc is not clear about `0` duration time. But maybe it is clear for others :)

**Comment by jerrinot** [comment]
I had another look and I'm interpreting it differently now:

There is indeed a null check in the `Duration` class: https://github.com/jsr107/jsr107spec/blob/916b3df8f45e5341748e9355ef7254bfccfb1b13/src/main/java/javax/cache/expiry/Duration.java#L108

I believe the check is there only because they do not want to throw NPE in the case the `durationAmount` is set to `0`. Because `timeUnit` is irrelevant in this case anyway.
 
JavaDoc says: "How long, in the specified units, the cache entries should live. 0 means eternal."  -> "If duration is set to 0 then it's eternal. Full Stop."  `TimeUnit` is not relevant in this case. 

However there is nothing indicating: "it's eternal if and only if `TimeUnit` is set to `null`" 
I skimmed the actual spec (not just JavaDoc), but I could not find any details there. 

The bottom line is: I agree with @alparslanavci our behaviour is buggy and we should interpret `0` as eternal duration. Even when `timeUnit` is non-null.

**Comment by serkan-ozal** [comment]
@jerrinot See the `Duration.ZERO`. It is used by TCK tests and it is not interpreted as eternal

**Comment by jerrinot** [comment]
Yes. You are right. This method makes it pretty clear: https://github.com/jsr107/jsr107spec/blob/916b3df8f45e5341748e9355ef7254bfccfb1b13/src/main/java/javax/cache/expiry/Duration.java#L188

**Comment by jerrinot** [comment]
I created https://github.com/jsr107/jsr107spec/issues/356
